### PR TITLE
Automodel: Filter unexploitable types in application mode.

### PR DESCRIPTION
**Pull request by max-schaefer**
We already did this in framework mode.

### CODE REVIEW

File: java/ql/automodel/src/AutomodelApplicationModeCharacteristics.qll [hunk 1]
```diff
@@ -28,25 +28,27 @@ newtype TApplicationModeEndpoint =
...
     AutomodelJavaUtil::isFromSource(call) and
     call = arg.getCall() and
-    idx = call.getCallee().getVaragsParameterIndex()
+    idx = call.getCallee().getVaragsParameterIndex() and
+    not AutomodelJavaUtil::isUnexploitableType(arg.getType())
   } or
-  TMethodReturnValue(Call call) {
+  TMethodReturnValue(MethodCall call) {
     AutomodelJavaUtil::isFromSource(call) and
-    not call instanceof ConstructorCall
```
Comment (max-schaefer) [inline java/ql/automodel/src/AutomodelApplicationModeCharacteristics.qll:45]: Drive-by fix: it's better to phrase this as a positive constraint.

**Comment by max-schaefer** [comment]
@tausbn, could I prevail on you to take a quick look at this?

**Review by tausbn — DISMISSED** [review]
Sorry for the late review! I started looking at this, and then got sidetracked and forgot to submit my review.

Overall I think this looks fine. 👍

**Comment by max-schaefer** [comment]
Thanks!

**Comment by max-schaefer** [comment]
@tausbn, could you re-approve, please?

**Pull request merged**